src/metabase/query_processor/api.clj

```diff
@@ -62,7 +62,8 @@
       (when-not database
         (throw (ex-info (tru "`database` is required for all queries whose type is not `internal`.")
                         {:status-code 400, :query query})))
-      (api/read-check :model/Database database))
+      ;; TODO FIXME!
```

Comment:
@metamben FYI this needs to be changed. We should check whether `can-read?` is true for a database, or there is a least one published table in this database that the current user has access to

Reply:
Done.